### PR TITLE
Adjust Navigation for PDF Export (EXPOSUREAPP-13438, EXPOSUREAPP-13439)

### PR DESCRIPTION
**Pull request by schauersbergern**
Quitting Export, Export errors and back navigation should all go back to personOverviewFragment
[Ticket](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-13438)
[Ticket](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-13439)

### CODE REVIEW

**Review by chiljamgossow — COMMENTED** [review]
Navigation from the dialogs are ok.
When pressing back from the pdf overview I would expect to see the info screen again, not the overview. Was that requested?

**Comment by SamuraiKek** [comment]
> Navigation from the dialogs are ok. When pressing back from the pdf overview I would expect to see the info screen again, not the overview. Was that requested?

I agree. That's how the flow works for single certs exports. I'm not sure whether this was requested or not, but either way, both cases should behave the same imo.

**Comment by schauersbergern** [comment]
> Navigation from the dialogs are ok. When pressing back from the pdf overview I would expect to see the info screen again, not the overview. Was that requested?

Talked with Kerstin Jumpertz and implemented it accordingly to keep backstack on backnav

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/pdf/ui/exportAll/DccExportAllOverviewFragment.kt [hunk 1]
```diff
@@ -30,13 +32,16 @@ class DccExportAllOverviewFragment : Fragment(R.layout.fragment_dcc_export_all_o
     @Inject
     lateinit var viewModelFactory: CWAViewModelFactoryProvider.Factory
     private val viewModel by cwaViewModels<DccExportAllOverviewViewModel> { viewModelFactory }
+    private val navOptions = NavOptions.Builder().setPopUpTo(R.id.personOverviewFragment, true).build()
```
Comment (chiljamgossow) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/pdf/ui/exportAll/DccExportAllOverviewFragment.kt:35]: You can move this into the method as well and only use a local variable

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/pdf/ui/exportAll/DccExportAllOverviewFragment.kt [hunk 2]
```diff
@@ -92,17 +97,21 @@ class DccExportAllOverviewFragment : Fragment(R.layout.fragment_dcc_export_all_o
             .setMessage(R.string.export_all_error_message)
             .setNeutralButton(R.string.export_all_error_faq) { _, _ ->
                 openUrl(R.string.certificate_export_all_error_dialog_faq_link)
-            }.setPositiveButton(android.R.string.ok) { _, _ -> }
-            .setOnDismissListener { popBackStack() }
+            }.setPositiveButton(android.R.string.ok) { _, _ -> popBackStack() }
```
Comment (chiljamgossow) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/pdf/ui/exportAll/DccExportAllOverviewFragment.kt:100]: Is this necessary?

**Review by chiljamgossow — COMMENTED** [review]
looks fine in general

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/pdf/ui/exportAll/DccExportAllOverviewFragment.kt (hunk 2 above)
Comment (schauersbergern) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/pdf/ui/exportAll/DccExportAllOverviewFragment.kt:101]: No, thanks for pointing out. Removed this call.

**Review by SamuraiKek — APPROVED** [review]
LGTM.

**Pull request merged**